radix-engine/src/ledger/traits.rs

```diff
@@ -124,6 +71,37 @@ pub trait SubstateStore {
         self.get_child_substate(address, &key)
             .map(|s| (s.value, s.phys_id))
     }
+
+    fn get_epoch(&self) -> u64;
+
+    // TODO: redefine what nonce is and how it's updated
+    // For now, we bump nonce only when a transaction has been committed
+    // or when an account is created (for testing).
+    fn get_nonce(&self) -> u64;
+}
+
+pub trait WriteableSubstateStore {
+    fn put_substate(&mut self, address: &[u8], substate: Substate);
+    fn put_child_substate(&mut self, address: &[u8], key: &[u8], substate: Substate);
+
+    fn put_keyed_substate(&mut self, address: &[u8], value: Vec<u8>, phys_id: (Hash, u32)) {
+        self.put_substate(address, Substate { value, phys_id });
+    }
```

Comment:
Didn't see the difference between `put_keyed_substate` and `put_substate`. 

I'm supposing that there will be some semantic difference in the next PR:
-  One uses conflict free substate id
-  The other uses derived substate id where existence check is required (if so, the interface should return a result).

Reply:
Yup, this is mostly temporary interface in fact.